clang/lib/Driver/Driver.cpp

```diff
@@ -1118,6 +1118,25 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
         << SYCLHostCompilerOptions->getSpelling().split('=').first
         << "-fsycl-host-compiler";
 
+  Arg *SYCLUIHArg =
+     C.getInputArgs().getLastArg(options::OPT_fsycl_use_integration_headers,
+                                 options::OPT_fno_sycl_use_integration_headers);
+  if (SYCLUIHArg) {
+    // -f[no-]sycl-use-integration-headers cannot be used without
+    // -fsycl or -fsycl-device-only
+    if (!HasValidSYCLRuntime)
+      Diag(clang::diag::err_drv_sycl_opt_requires_opt)
+          << "-f[no-]sycl-use-integration-headers";
+    // -fno-sycl-use-integration-headers cannot be used with
+    // -fsycl-host-compiler
+    if (SYCLHostCompiler &&
+      SYCLUIHArg->getOption().matches(
+          options::OPT_fno_sycl_use_integration_headers))
+      Diag(clang::diag::err_drv_option_conflict)
+          << SYCLHostCompiler->getSpelling()
+          << "-fno-sycl-use-integration-headers";
```

Comment:
Similar diagnostics above strip the `=argument` from the `SYCLHostCompiler` spelling. Perhaps we should do similarly here.
```suggestion
      Diag(clang::diag::err_drv_option_conflict)
          << SYCLHostCompiler->getSpelling().split('=').first
          << "-fno-sycl-use-integration-headers";
```

Reply:
Done, thanks.